diff --git a/cmake/onnxruntime_unittests.cmake b/cmake/onnxruntime_unittests.cmake
index b23365e..58e60af 100644
--- a/cmake/onnxruntime_unittests.cmake
+++ b/cmake/onnxruntime_unittests.cmake
@@ -1612,7 +1612,7 @@ if (NOT CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
   endif()
 
   if (NOT onnxruntime_ENABLE_TRAINING_TORCH_INTEROP)
-    if (onnxruntime_BUILD_JAVA AND NOT onnxruntime_ENABLE_STATIC_ANALYSIS)
+    if (onnxruntime_BUILD_JAVA AND NOT onnxruntime_ENABLE_STATIC_ANALYSIS AND NOT CMAKE_SYSTEM_NAME STREQUAL "Android")
       block()
         message(STATUS "Enabling Java tests")
 
